### DISCUSSION TITLE
Get x,y,z coordinates for the peak e-field

### DISCUSSION THREAD

**Discussion by akelkar26** [simnibs/simnibs#195] [9. Results and analysis] [answered]
### Discussed in https://github.com/simnibs/simnibs/discussions/193

<div type='discussions-op-text'>

<sup>Originally posted by **akelkar26** September  5, 2023</sup>
Hi, 

Hope you are well. Thank you for this great software!
I am trying to get the x,y,z coordinates for the maximum (peak) e-field value. I couldn't find much documentation on how to get that. Could you please help?

Also, would you suggest using the 99.9% magnE value as the peak e-field value, to avoid outliers?

Awaiting your reply.
Thank you
Best,
Apoorva </div>

**Comment by TorgeW** [comment, marked as answer]
```python
import numpy as np
from simnibs import Msh

m = Msh(fn='path/to/sim.msh')

# Select gra matter and white matter tissues
tissues = [1,2]
m = m.crop_mesh(tags=tissues)

tetrahedra_centers = m.elements_baricenters()[:]
tetrahedra_emag = m.field['magnE'][:]

percentile_mask = tetrahedra_emag < np.percentile(tetrahedra_emag, 99.99)

max_emag_index = np.argmax(tetrahedra_emag[percentile_mask])
max_emag_coord = tetrahedra_centers[percentile_mask][max_emag_index]

print(tetrahedra_emag[percentile_mask][max_emag_index], max_emag_coord)
```

This should give you the peak e-field position. 
Using percentiles to avoid outliers is a good idea, you might have to experiment a bit to find the perfect percentile.

**Comment by akelkar26** [comment]
Thank you so much! That worked perfectly on my data.
